source/adapters/level_zero/common.cpp

```diff
@@ -332,15 +332,21 @@ template <> zes_structure_type_t getZesStructureType<zes_mem_properties_t>() {
 
 // Global variables for ZER_EXT_RESULT_ADAPTER_SPECIFIC_ERROR
 thread_local ur_result_t ErrorMessageCode = UR_RESULT_SUCCESS;
-thread_local char ErrorMessage[MaxMessageSize];
+thread_local char ErrorMessage[MaxMessageSize]{};
 thread_local int32_t ErrorAdapterNativeCode;
 
 // Utility function for setting a message and warning
 [[maybe_unused]] void setErrorMessage(const char *pMessage,
                                       ur_result_t ErrorCode,
                                       int32_t AdapterErrorCode) {
-  assert(strlen(pMessage) <= MaxMessageSize);
-  strcpy(ErrorMessage, pMessage);
+  assert(strlen(pMessage) < MaxMessageSize);
+  // Copy at most MaxMessageSize - 1 bytes to ensure the resultant string is
+  // always null terminated.
+#if defined(_WIN32)
+  strncpy_s(ErrorMessage, MaxMessageSize - 1, pMessage, strlen(pMessage));
```

Comment:
tbh I'd just ignore coverity complaining about not using safe windows string functions, and use `strncpy` on all platforms. The `strlen(pMessage)` does not make this any safer.

Reply:
I copied this from other places with strcpy ([here](https://github.com/oneapi-src/unified-runtime/blob/main/source/loader/ur_print.cpp#L33) and [here](https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/mock/ur_mock.cpp#L38)), I remember there being issues on some windows configs with warnings being generated if you don't use strncpy_s. I didn't actually check if that was still a concern, happy to delete if it isn't

Reply:
> tbh I'd just ignore coverity complaining about not using safe windows string functions, and use `strncpy` on all platforms.

I'm happy with this approach.

Reply:
> I copied this from other places with strcpy ([here](https://github.com/oneapi-src/unified-runtime/blob/main/source/loader/ur_print.cpp#L33) and [here](https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/mock/ur_mock.cpp#L38)), I remember there being issues on some windows configs with warnings being generated if you don't use strncpy_s. I didn't actually check if that was still a concern, happy to delete if it isn't

Well, the theory is that if you know the length of the string (which we don't here), you can supply it to `strncpy_s` and it can do bounds checking without having to iterate over the cstring. This protects against malformed cstrings that are not 0-terminated.

Reply:
deleted the strncpy_s path